vcd/datasource_vcd_nsxt_edgegateway_static_route.go

```diff
@@ -0,0 +1,167 @@
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge gateway ID for Static Route configuration",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of Static Route",
+			},
+			"network_cidr": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Network CIDR (e.g. 192.168.1.1/24) for Static Route",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of Static Route",
+			},
+			"next_hop": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set of next hops used within the static route",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_address": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "IP Address of next hop",
+						},
+						"admin_distance": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Admin distance of next hop",
+						},
+						"scope": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "ID of Scope element",
+									},
+									"name": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Name of Scope element",
+									},
+									"type": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Scope type - One of 'NETWORK', 'SYSTEM_OWNED'",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func datasourceVcdNsxtEdgeGatewayStaticRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	orgName := d.Get("org").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	nsxtEdge, err := vcdClient.GetNsxtEdgeGatewayById(orgName, edgeGatewayId)
+	if err != nil {
+		return diag.Errorf("[NSX-T Edge Gateway Static Route DS read] error retrieving Edge Gateway: %s", err)
+	}
+
+	searchName := d.Get("name").(string)
+	searchNetworkCidr := d.Get("network_cidr").(string)
+
+	// API does not support filtering yet therefore all Static Routes need to be retrieved and filtered manually
+	// There is also a problem that 'name' must not be unique as well as 'network_cidr', however the
+	// goal is to attempt 2 pass filtering - 'name' is mandatory, but if it happens it is not
+	// unique, one can attempt to make the search more precise by specifying 'network_cidr'
+	allStaticRoutes, err := nsxtEdge.GetAllStaticRoutes(nil)
```

Comment:
Just throwing the idea of refactoring this into a go-vcloud-director: `nsxtEdge.GetStaticRouteByName(name, cidr)`
No need to make the huge refactor now, but just an idea.

Reply:
I did think about this when implementing it, but this looked like a sort of nonstandard approach to the problem. Using SDK, one could think of other ways as well.